clang/lib/CodeGen/BackendUtil.cpp

```diff
@@ -1141,7 +1141,7 @@ void EmitAssemblyHelper::RunOptimizationPipeline(
       // used only with spir triple.
       if (CodeGenOpts.SPIRITTAnnotations) {
         assert(
-            TargetTriple.isSPIR() &&
+            (TargetTriple.isSPIR() || TargetTriple.isSPIRV()) &&
```

Comment:
For all these if checks, could we make a utility function here that is like isSupportSPIRTriple or something that does this, instead of having the same || check everything?

Reply:
Thanks Nick. I think that is a good idea. Maybe something simpler like isSPIROrSPIRV?

Reply:
Sure, whatever you think coveys the idea.

Reply:
Thanks @sarnex. I have added a utility function to help and have replaced almost all places (except places with Mike Toguchi pointed out that we need to stick to just isSPIR check).